File: conn/ado/TrafodionProvider/Trafodion.Data/Network/Messages/FetchReply.cs
/**********************************************************************
// @@@ START COPYRIGHT @@@
//
// (C) Copyright 2009-2015 Hewlett-Packard Development Company, L.P.
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
//
// @@@ END COPYRIGHT @@@
********************************************************************/

using System;

namespace Trafodion.Data
{
    internal class FetchReply: INetworkReply
    {
        public ReturnCode returnCode;
        public SqlWarningOrError[] errorList;
        public long rowsAffected;
        public int dataFormat;
        public int dataOffset;

        public void ReadFromDataStream(DataStream ds, TrafodionDBEncoder enc)
        {
            returnCode = (ReturnCode)ds.ReadInt32();

            if (returnCode != ReturnCode.Success && returnCode != ReturnCode.NoDataFound)
            {
                errorList = SqlWarningOrError.ReadListFromDataStream(ds, enc);
            }
            else
            {
                errorList = new SqlWarningOrError[0];
            }

            rowsAffected = ds.ReadUInt32();
            dataFormat = ds.ReadInt32();

            ds.ReadInt32(); // ignore length since we save entire buffer
            dataOffset = ds.Position;

            /*if (returnCode == ReturnCode.Success || returnCode == ReturnCode.SuccessWithInfo)
            {
                data = ds.ReadBytes();
            }*/
        }
    }
}

File: core/conn/jdbc_type2/src/main/java/org/trafodion/jdbc/t2/FetchReply.java
/*******************************************************************************
// @@@ START COPYRIGHT @@@
//
// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.
//
// @@@ END COPYRIGHT @@@
 *******************************************************************************/

package org.trafodion.jdbc.t2;

import java.nio.charset.CharacterCodingException;
import java.nio.charset.UnsupportedCharsetException;

class FetchReply {
    int returnCode;
    int totalErrorLength;
    SQLWarningOrError[] errorList;
    int rowsAffected;
    int outValuesFormat;
    byte[] outValues;

    // -------------------------------------------------------------
    public FetchReply(LogicalByteArray buf, InterfaceConnection ic) throws CharacterCodingException,
           UnsupportedCharsetException {
               buf.setLocation(0);

               returnCode = buf.extractInt();

               if (returnCode != TRANSPORT.SQL_SUCCESS && returnCode != TRANSPORT.NO_DATA_FOUND) {
                   totalErrorLength = buf.extractInt();
                   if (totalErrorLength > 0) {
                       errorList = new SQLWarningOrError[buf.extractInt()];
                       for (int i = 0; i < errorList.length; i++) {
                           errorList[i] = new SQLWarningOrError(buf, ic, InterfaceUtilities.SQLCHARSETCODE_UTF8);
                       }
                   }
               }

               if (errorList == null) {
                   errorList = new SQLWarningOrError[0];
               }

               rowsAffected = buf.extractInt();
               outValuesFormat = buf.extractInt();

               if (returnCode == TRANSPORT.SQL_SUCCESS || returnCode == TRANSPORT.SQL_SUCCESS_WITH_INFO) {
                   outValues = buf.extractByteArray();
               }
    }
}
